osu.Game/Screens/Import/ReplayMissingBeatmapScreen.cs

```diff
@@ -0,0 +1,225 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Audio;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.Drawables.Cards;
+using osu.Game.Configuration;
+using osu.Game.Database;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Overlays;
+using osu.Game.Overlays.Settings;
+using osu.Game.Scoring;
+using osu.Game.Screens.Ranking;
+using osuTK;
+using Realms;
+
+namespace osu.Game.Screens.Import
+{
+    [Cached(typeof(IPreviewTrackOwner))]
+    public partial class ReplayMissingBeatmapScreen : OsuScreen, IPreviewTrackOwner
+    {
+        [Resolved]
+        private BeatmapModelDownloader beatmapDownloader { get; set; } = null!;
+
+        [Resolved]
+        private IAPIProvider api { get; set; } = null!;
+
+        [Resolved]
+        private ScoreManager scoreManager { get; set; } = null!;
+
+        [Resolved]
+        private RealmAccess realm { get; set; } = null!;
+
+        private IDisposable? realmSubscription;
+
+        [Cached]
+        private OverlayColourProvider colourProvider = new OverlayColourProvider(OverlayColourScheme.Purple);
+
+        [Resolved]
+        private INotificationOverlay? notificationOverlay { get; set; }
+
+        private Container beatmapPanelContainer = null!;
+        private ReplayDownloadButton replayDownloadButton = null!;
+        private SettingsCheckbox automaticDownload = null!;
+
+        private readonly MemoryStream? scoreStream;
+        private readonly APIBeatmap beatmap;
+
+        private APIBeatmapSet? beatmapSetInfo;
+
+        public ReplayMissingBeatmapScreen(APIBeatmap beatmap, MemoryStream? scoreStream = null)
```

Comment:
Why is `scoreStream` allowed to be null if nothing ever uses that?

Reply:
becuase `archive.GetStream()` retrun a nullable stream, so I make this nullable.
I will remove it.